### DISCUSSION TITLE
Old vs New Velocity Tutorial

### DISCUSSION THREAD

**Discussion by ACastanza** [COMBINE-lab/salmon#719] [Q&A (single-cell specific)]
Hello @rob-p and all,

I have a couple questions about the [old](https://combine-lab.github.io/alevin-tutorial/2020/alevin-velocity/) vs [new](https://combine-lab.github.io/alevin-fry-tutorials/2021/alevin-fry-velocity/) RNA-velocity with Alevin tutorials.

After reading through them several times, and having fully implemented the old tutorial into a pipeline, I have a couple questions about converting.

The major benefit of the new tutorial seems to be that alevin-fry produces the "usa mode" result when fed a three column tx2gene map file which is then simple to parse into the h5ad spec expected by scanpy/scvelo. With the previous tutorial, instead there was a two column tx2gene map with -I suffixes to indicate the unspliced sequences. **Is it possible to get standard alevin to produce the "usa mode" outputs without fully converting to alevin-fry**?
(If not, I'd like to propose this as a feature request)

Additionally, it appears that the flank length has been changed from read length -1 in the old tutorial to read length -5 in the new tutorial. **Is there a particular significance to this change?**

The new tutorial uses a simple "splici reference" and no longer adds the full genome as a decoy sequence. **Is is no longer advisable to use the full decoy or partial decoy indices when computing with either alevin or alevin-fry** (assuming that speed/memory are not the highest concern)? **If adding the decoys is not contraindicated, is it possible to run the new tutorial pipeline with an index which has decoy sequences added to it ala the old tutorial?**

Edit: I see that decoys are not compatible with alevin-fry mode? 
> Automatically detect and exit if alevin is run with an index including decoy sequences when using the --rad and/or --sketch flags. This functionality is not currently supported, and mapping against such an index can cause (cryptic) errors in downstream processing. Now, if such an index is passed when using these flags, an informative error message is printed and the program will exit with a return code of 1.

Parsing the USA mode file combines the Spliced and Ambiguous reads into a single layer rather than assigning the ambiguous reads to the expected ambiguous layer. Would it not be better to assign the ambiguous reads to the ambiguous layer (which tools like scvelo expect) and then let scvelo handle the interpretation of this layer? Combining S+A "early" seems to negate the benefit of the new USA mode over the old tutorial where alevin was allowed to assign these ambiguous reads as it saw fit (learning the expected proportions from the data). Assuming my understanding is correct here **would it be possible to get instructions for extracting the three layers separately so that they can be assembled into the expected three layer h5ad file?**

Since the previous velocity tutorial explicitly requested a decoy aware index and the new alevin-fry mode doesn't support this. It seems like an option to produce the USA mode result file from standard alevin would provide a "best of both worlds" situation.

Edit: additional question:
When making the intron aware transcritpome fasta the "joinOverlappingIntrons" parameter in `getFeatureRanges` has been changed to `TRUE` where it was previously `FALSE`, is there a particular reason for this change?

I'm sure I'll have some more questions as I work through the new velocity tutorial but that's all I've got for now, thanks!

**Comment by wmacnair** [comment]
Hey, just bumping this to show wider interest :)

What I would like is: an accurate, fast (ish), streamlined pipeline for mapping to unspliced / spliced / ambiguous reads. So the question that is most interesting to me is **understanding how does decoy mode vs splici mode affect accuracy and timings**. 

It would be great if you could comment on this! 

Thanks, Will

**Comment by wmacnair** [comment]
I've just discovered that these questions were answered in the `alevin-fry` repo, here:
https://github.com/COMBINE-lab/alevin-fry/issues/31
